src/util/cilfacade.ml

```diff
@@ -80,71 +80,45 @@ let end_basic_blocks f =
   let thisVisitor = new allBBVisitor in
   visitCilFileSameGlobals thisVisitor f
 
+class removeLabelsVisitor = object
+  inherit nopCilVisitor
+
+  val nestingDepth = ref 0
```

Comment:
I think the more canonical way would be `mutable val nestingDepth = 0`, which needs no dereferencing and can be updated with `<-` just like mutable record fields.

Reply:
Yes, I always forget what is syntactic sugar for what mutable record field for references or the other way around.